engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/ExecutionContextManager.scala

```diff
@@ -90,6 +90,20 @@ class ExecutionContextManager {
       contexts.contains(contextId)
     }
 
+  /** Set oneshot expression for the specified context.
+    *
+    * @param contextId the identifier of the execution context
+    * @param oneshotExpression the oneshot visualization
+    */
+  def setOneshotExpression(
+    contextId: ContextId,
+    oneshotExpression: OneshotExpression
+  ): Unit =
+    synchronized {
+      val state = contexts(contextId)
+      state.visualizations.setOneshotExpression(oneshotExpression)
```

Comment:
Can there be only one `oneshotExpression`? That seems suspicious to me.

Especially when (one shot) executions requests are co-leased the `oneshotExpression`s can accumulate - one the (last) execution finishes, all the `oneshotExpression`s need to be evaluated and the results delivered to the IDE, I think.